app/notification.go

```diff
@@ -349,76 +349,156 @@ func sendNotificationEmail(post *model.Post, user *model.User, channel *model.Ch
-	day := fmt.Sprintf("%d", tm.Day())
-	year := fmt.Sprintf("%d", tm.Year())
-	zone, _ := tm.Zone()
+	go func() {
+		if err := utils.SendMail(user.Email, html.UnescapeString(subjectText), bodyText); err != nil {
+			l4g.Error(utils.T("api.post.send_notifications_and_forget.send.error"), user.Email, err)
+		}
+	}()
 
-	if channel.Type == model.CHANNEL_DIRECT {
-		bodyText = userLocale("api.post.send_notifications_and_forget.message_body")
-		subjectText = userLocale("api.post.send_notifications_and_forget.message_subject")
+	if einterfaces.GetMetricsInterface() != nil {
+		einterfaces.GetMetricsInterface().IncrementPostSentEmail()
+	}
 
-		senderDisplayName := senderName
+	return nil
+}
 
-		mailTemplate = "api.templates.post_subject_in_direct_message"
-		mailParameters = map[string]interface{}{"SubjectText": subjectText,
-			"SenderDisplayName": senderDisplayName, "Month": month, "Day": day, "Year": year}
-	} else if channel.Type == model.CHANNEL_GROUP {
-		bodyText = userLocale("api.post.send_notifications_and_forget.mention_body")
+/**
+ * Computes the subject line for direct notification email messages
+ */
+func getDirectMessageNotificationEmailSubject(post *model.Post, translateFunc i18n.TranslateFunc, siteName string, senderName string) string {
+	_, month, day, year, _ := getFormattedPostTime(post, translateFunc)
+	var subjectParameters = map[string]interface{}{
+		"SiteName":          siteName,
+		"SenderDisplayName": senderName,
+		"Month":             month,
+		"Day":               day,
+		"Year":              year,
+	}
+	return translateFunc("app.notification.subject.direct.full", subjectParameters)
+}
 
-		senderDisplayName := senderName
+/**
+ * Computes the subject line for group, public, and private email messages
+ */
+func getNotificationEmailSubject(post *model.Post, translateFunc i18n.TranslateFunc, siteName string, teamName string) string {
+	_, month, day, year, _ := getFormattedPostTime(post, translateFunc)
+	var subjectParameters = map[string]interface{}{
+		"SiteName": siteName,
+		"TeamName": teamName,
+		"Month":    month,
+		"Day":      day,
+		"Year":     year,
+	}
+	return translateFunc("app.notification.subject.notification.full", subjectParameters)
+}
 
-		mailTemplate = "api.templates.post_subject_in_group_message"
-		mailParameters = map[string]interface{}{"SenderDisplayName": senderDisplayName, "Month": month, "Day": day, "Year": year}
-		channelName = userLocale("api.templates.channel_name.group")
+/**
+ * Computes the email body for notification messages
+ */
+func getNotificationEmailBody(recipient *model.User, post *model.Post, channel *model.Channel, senderName string, teamName string, teamURL string, emailNotificationContentsType string, translateFunc i18n.TranslateFunc) string {
+	// only include message contents in notification email if email notification contents type is set to full
+	var bodyPage *utils.HTMLTemplate
+	if emailNotificationContentsType == model.EMAIL_NOTIFICATION_CONTENTS_FULL {
```

Comment:
This needs to check the license. Also, I'd lean towards having generic contents as the specific cause with full contents being the generic one, but that shouldn't really matter if we do proper validation of the config